plugins/modules/organization.py

```diff
@@ -87,11 +171,50 @@ def main():
             name=dict(required=True),
             description=dict(),
             label=dict(),
+            upstream_type=dict(required=False, choices=['redhat_cdn', 'export_sync', 'network_sync']),
+            upstream_url=dict(required=False),
+            upstream_username=dict(required=False),
+            upstream_password=dict(required=False, no_log=True),
+            # upstream_ca_cert=dict(required=False, type='entity', resource_type='content_credentials', scope=['']),
+            upstream_ca_cert_id=dict(required=False),
+            upstream_organization=dict(required=False),
+            upstream_lifecycle_environment=dict(required=False),
+            upstream_content_view=dict(required=False),
         ),
     )
 
     with module.api_connection():
-        module.run()
+        handle_cdn_configuration = 'upstream_type' in module.foreman_params
+
+        organization = module.lookup_entity('entity')
+        new_organization = module.run()
+
+        if handle_cdn_configuration:
+            if organization:
+                payload = {
+                    'id': new_organization['id'],
+                    'type': module.foreman_params['upstream_type'],
+                }
+                extra_payload = {}
+
+                if module.foreman_params['upstream_type'] == 'redhat_cdn':
+                    extra_payload = {
+                        'url': module.foreman_params['upstream_url'],
+                    }
+                if module.foreman_params['upstream_type'] == 'network_sync':
+                    extra_payload = {
+                        'url': module.foreman_params['upstream_url'],
+                        'ssl_ca_credential_id': module.foreman_params['upstream_ca_cert_id'],
+                        'username': module.foreman_params['upstream_username'],
+                        'password': module.foreman_params['upstream_password'],
+                        'upstream_organization_label': module.foreman_params['upstream_organization'],
+                        'upstream_lifecycle_environment_label': module.foreman_params['upstream_lifecycle_environment'],
+                        'upstream_content_view_label': module.foreman_params['upstream_content_view'],
+                    }
+
+                if extra_payload:
+                    payload.update(extra_payload)
+                module.resource_action('organizations', 'cdn_configuration', payload)
```

Comment:
Do I read it right, this essentially *always* returns `changed=True` now, as there is no way to "diff" the cdn config?!

Reply:
The data is shown in the API reply to `/katello/api/organizations/:id`, so we *could* diff it… What a mess

Reply:
I have tried poking around in the API, but I can't find it.

When I go to ```https://sat.rh.lab/api/organizations/1/cdn_configuration``` I get a 404. And it's not listed in the 'overview' at the ```organizations/<org_id>``` endpoint either.

Reply:
attention attention! katello overrides the org controller!

you need to go to `/katello/api/organizations/:id`, not to `/api/organizations/:id`, the apipie library will do the right thing for us.

and yes, the `cdn_configuration` part is only defined for `PUT` (update) operations, you can't `GET` it, but the data is part of the normal org show view, if you look at the right controller

Reply:
I feel my Python-Fu is lacking a bit, but I made an attempt to compare some things, and it seems to work well :-)

This will never be idempotent for ```network_sync``` as the current password cannot be compared with the provided password.